src/main/scala/akka/persistence/cassandra/journal/StreamMergerActor.scala

```diff
@@ -112,6 +119,8 @@ private class StreamMergerActor(
           case MergeFailure(_, _, _, _) =>
             merging(journalIdProgress, persistenceIdProgress, step + 50l)
           case MergeSuccess(newJournalIdProgress, newPersistenceIdProgress, mergedStream) =>
+            writeIndexProgress(mergedStream)
+            writeProgress(newJournalIdProgress, newPersistenceIdProgress)
             merging(newJournalIdProgress, newPersistenceIdProgress, step)
```

Comment:
With asynchronous write operations, shouldn't we continue `merging` only after the writes successfully complete?